NumLib/NewtonMethods/NewtonRaphson.h

```diff
@@ -0,0 +1,79 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_NEWTONMETHODS_NEWTONRAPHSON_H_
+#define NUMLIB_NEWTONMETHODS_NEWTONRAPHSON_H_
+
+#include <logog/include/logog.hpp>
+
+#include <Eigen/Dense>
+
+namespace NumLib
```

Comment:
Moving this to MaterialLib/SolidModels because the implementation operates on implicit state which is updated with `update_solution(increment)`, instead of `solution += increment` in the loop body.